src/lib.rs

```diff
@@ -261,7 +261,11 @@ macro_rules! into_impl {
 
 impl<S: AsRef<str>> From<S> for UniCase<S> {
     fn from(s: S) -> Self {
-        UniCase::unicode(s)
+        if s.as_ref().is_ascii() {
```

Comment:
Perhaps we just call `UniCase::new(s)` here? Then we can be sure the code is always the same!

If it's not too much of a hassle. If it is, I can merge as-is.

Reply:
That's a good idea and not a hassle at all! I'll push up

Reply:
Done in https://github.com/seanmonstar/unicase/pull/77/commits/10383d1ac38699faf358c76347e7e0a03fd85e8a